ui/components/src/LineChart/LineChart.tsx

```diff
@@ -112,6 +134,30 @@ export function LineChart({
   const [isDragging, setIsDragging] = useState(false);
   const [startX, setStartX] = useState(0);
 
+  useImperativeHandle(
```

Comment:
I think exposing these as imperative handles is better than trying to do something with props and useEffect here, which would likely be buggy and error prone and might exacerbate some of the re-render perf issues we have. Let me know if you disagree and have thoughts on an alternative.

Reply:
Really cool!